tutorial/practicals/LFRic/building_code/1_simple_kernels/part2/README.md

```diff
@@ -56,34 +56,72 @@ a generic discontinuous function space `ANY_DISCONTINUOUS_SPACE_n`
 https://psyclone.readthedocs.io/en/stable/dynamo0p3.html#supported-function-spaces)
 for information on the supported function spaces).
 
-Unlike the specific function space identifier `W0` that is located
+* Unlike the specific function space identifier `W0` that is located
 in the LFRic infrastructure module `fs_continuity_mod`, the metadata
 identifiers for the generic function spaces are located in
-the LFRic infrastructure module `argument_mod`. We will need to remove
-the `use fs_continuity_mod, only: W0` statement and add the generic
-`ANY_SPACE_1` metadata identifier to the `use argument_mod, only: ...`
-statement.
+the LFRic infrastructure module `argument_mod`, so the statement
+`use argument_mod, only: ...` is modified accordingly.
+
+---
 
-After that we need to change the kernel code unit names accordingly,
-e.g. `setval_field_w0_` becomes `setval_field_any_`, and then we
-will run the kernel stub generator on the renamed kernel
+The kernel `setval_field_any_code()` subroutine body is empty and needs
+to be populated. As in [Part 1](../pasrt1), we will first create the code
+for the argument list and declarations by running the PSyclone kernel
+stub generator:
 
 ```shell
 genkernelstub setval_field_any_kernel_mod.f90
 ```
 
-to check and update the kernel argument names and order if required.
+and then replace the empty kernel subroutine body with the generated
+one that contains the argument list and the generated declarations.
+Again, the generated `USE constants_mod, ONLY: r_def, i_def` statement
+needs to be removed as the externally defined parameters are already
+listed in the `use` statements at the kernel module level.
+
+---
+**NOTE**
+
+The argument list and declarations are generated in one line per
+statement each, so in this case of long argument names they may overrun
+the free-form Fortran line-length limit of 132 characters and PSyclone
```

Comment:
They can just use the `-l` flag to `genkernelstub`.

Reply:
Rephrased, thanks.